### PR TITLE
chore(all): reduce cognitive complexity

### PR DESCRIPTION
**Pull request by biffgaut**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: f64216eaab9fd15209a1bbb066c919a690a3e081
* Result: SUCCEEDED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=2fe6ee43-b447-4030-a269-beddedac4d88%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: githubautobuild-for-cdk-v2
* Commit ID: 8e5b41021600b9af52c22f77f4b404b4cbc735bb
* Result: SUCCEEDED
* [Build Logs](https://t2ulkoluaf.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=2fa196b9-9e40-44aa-b431-ab33158ccd30%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/aws-openapigateway-lambda/lib/index.ts [hunk 1]
```diff
@@ -164,6 +164,9 @@ export class OpenApiGatewayToLambda extends Construct {
     let lambdaCounter = 0;
 
     this.apiLambdaFunctions = props.apiIntegrations.map(apiIntegration => {
+      if (apiIntegration.existingLambdaObj && apiIntegration.lambdaFunctionProps) {
```
Comment (georgebearden) [inline source/patterns/@aws-solutions-constructs/aws-openapigateway-lambda/lib/index.ts:167]: I like where you placed the error checking, but this is also done in the else statement below - see [here](https://github.com/awslabs/aws-solutions-constructs/blob/main/source/patterns/%40aws-solutions-constructs/aws-openapigateway-lambda/lib/index.ts#L176)

**Review by georgebearden — APPROVED** [review]
looks good to me, just one comment around a redundant check.

**Pull request merged**